services/checkin/controller/controller.go

```diff
@@ -62,30 +62,30 @@ func CreateUserCheckin(w http.ResponseWriter, r *http.Request) {
 	can_user_checkin, err := service.CanUserCheckin(user_checkin.ID, user_checkin.Override)
 
 	if err != nil {
-		panic(errors.UnprocessableError(err.Error()))
+		panic(errors.InternalError(err.Error(), "Unable to determine user's check-in permissions."))
 	}
 
 	if !can_user_checkin {
-		panic(errors.UnprocessableError("Attendee must be RSVPed to check-in (or have a staff override)."))
+		panic(errors.AttributeMismatchError(err.Error(), "Attendee must be RSVPed to check-in (or have a staff override)."))
 	}
 
 	err = service.CreateUserCheckin(user_checkin.ID, user_checkin)
 
 	if err != nil {
-		panic(errors.UnprocessableError(err.Error()))
+		panic(errors.DatabaseError(err.Error(), "Could not create user check-in."))
 	}
 
 	updated_checkin, err := service.GetUserCheckin(user_checkin.ID)
 
 	if err != nil {
-		panic(errors.UnprocessableError(err.Error()))
+		panic(errors.DatabaseError(err.Error(), "Could not get recently created check-in information."))
 	}
 
 	if updated_checkin.Override {
 		err = service.AddAttendeeRole(updated_checkin.ID)
 
 		if err != nil {
-			panic(errors.UnprocessableError(err.Error()))
+			panic(errors.InternalError(err.Error(), "Could not add attendee role to user.\n"+err.Error()))
```

Comment:
Don't need a newline here and don't need to append `err.Error()` since it will be included in the `RawError` field.